src/Microsoft.FeatureManagement/ServiceCollectionExtensions.cs

```diff
@@ -26,11 +27,21 @@ public static IFeatureManagementBuilder AddFeatureManagement(this IServiceCollec
             // Add required services
             services.TryAddSingleton<IFeatureDefinitionProvider, ConfigurationFeatureDefinitionProvider>();
 
-            services.AddSingleton<IFeatureManager, FeatureManager>();
+            services.TryAddSingleton<IFeatureVariantOptionsResolver, ConfigurationFeatureVariantOptionsResolver>();
 
-            services.AddSingleton<ISessionManager, EmptySessionManager>();
+            services.TryAddSingleton<IContextualFeatureVariantAssigner<ITargetingContext>, ContextualTargetingFeatureVariantAssigner>();
```

Comment:
Is this supposed to be added by the user code?

Reply:
Yes, thank you!